maven-core/src/main/java/org/apache/maven/bridge/MavenRepositorySystem.java

```diff
@@ -654,7 +654,7 @@ public ArtifactRepository createLocalRepository(MavenExecutionRequest request, F
 
     public ArtifactRepository createLocalRepository(File localRepository) throws InvalidRepositoryException {
         return createRepository(
-                "file://" + localRepository.toURI().getRawPath(),
+                localRepository.toPath().toUri().toASCIIString(),
```

Comment:
This change seems unrelated, will try to undo this, as IMHO is actually not needed.